server/graphql/v1/mutations/collectives.js

```diff
@@ -341,52 +340,41 @@ export function editCollective(_, args, req) {
 
   let collective, parentCollective;
 
-  const promises = [
-    req.loaders.collective.findById.load(args.collective.id).then(c => {
-      if (!c) throw new Error(`Collective with id ${args.collective.id} not found`);
+  return req.loaders.collective.findById
+    .load(args.collective.id)
+    .then(c => {
+      if (!c) {
+        throw new Error(`Collective with id ${args.collective.id} not found`);
+      }
       collective = c;
-    }),
-  ];
-
-  if (args.collective.ParentCollectiveId) {
-    promises.push(
-      req.loaders.collective.findById.load(args.collective.ParentCollectiveId).then(pc => {
-        if (!pc)
-          return Promise.reject(new Error(`Parent collective with id ${args.collective.ParentCollectiveId} not found`));
-        parentCollective = pc;
-      }),
-    );
-  }
-  return Promise.all(promises)
+    })
     .then(() => {
-      // If trying to edit parent collective, ensure user is an admin of it
-      const newParentCollectiveId = args.collective.ParentCollectiveId;
-      const isChangingParent = newParentCollectiveId && newParentCollectiveId !== collective.ParentCollectiveId;
-      if (isChangingParent && !req.remoteUser.isAdmin(newParentCollectiveId)) {
-        throw new errors.Unauthorized({
-          message: `You must be logged in as the creator of this Event or as an admin of the ${parentCollective.slug} collective to edit this Event Collective`,
+      if (collective.ParentCollectiveId) {
+        return req.loaders.collective.findById.load(collective.ParentCollectiveId).then(pc => {
+          if (!pc) {
```

Comment:
Not sure I would throw an exception here.

If the ParentCollectiveId is invalid, then we can't edit the Collective anymore.

It's not supposed to happen, but I don't think there is a point checking that here.

Reply:
Not supposed to happen indeed but I'd rather have this check in place, otherwise the behaviour when `ParentCollectiveId` is incorrect will be harder to predict: `isAdmin` will return false in https://github.com/opencollective/opencollective-api/blob/94379ff62f64b5f36ccbdef45ff8fc8045855508/server/graphql/v1/mutations/collectives.js#L369 but you have to know (or go down the code) to understand that `isAdmin(null) === false`

Reply:
Ok. This needs a strong refactor anyway (the pattern to load `collective` and `parentCollective` is ugly) so that will certainly be tackled at this time.